libraries/AP_L1_Control/AP_L1_Control.cpp

```diff
@@ -238,7 +238,7 @@ void AP_L1_Control::update_waypoint(const struct Location &prev_WP, const struct
     // Calculate time varying control parameters
     // Calculate the L1 length required for specified period
     // 0.3183099 = 1/1/pipi
-    _L1_dist = 0.3183099f * _L1_damping * _L1_period * groundSpeed;
+    _L1_dist = MAX(0.3183099f * _L1_damping * _L1_period * groundSpeed, dist_min);
```

Comment:
Are we clamping an intermediate value where we should be clamping something further down the calculation?

Reply:
good question that I hadn't considered before.

There's only one other place lower down that could be affected and this when the vehicle has not reached the origin (aka WP_A) and it decides whether to head for the origin or try to get on the track from origin to destination.  This extra clamp will mean it needs to be outside the waypoint radius to head for the origin which makes sense to me.